core/arch/interp.c

```diff
@@ -3377,13 +3381,15 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             bb->instr_start = bb->cur_pc;
             if (bb->full_decode) {
                 /* only going through this do loop once! */
-                bb->cur_pc = decode(dcontext, bb->cur_pc, bb->instr);
+                bb->cur_pc = IF_AARCH64_ELSE(decode_with_ldstex, decode)
```

Comment:
suggest: IMHO it would be better to have the option check outside, or to rename the routine.  As is, this looks like it's going to decode a "ldstex" no matter what.  (I still am not a fan of this decode-bundle-inside-decode-routine approach: I would rather see the regular decode loop here with decisions regarding multiple instructions made at the multi-instruction level.)